functions/src/models/requests/timeline.ts

```diff
@@ -168,9 +168,9 @@ export class TimelineItem implements ITimelineItem {
 
   toObject(): object {
     return {
-      actorRef: this.actorRef,
-      offerRef: this.offerRef,
-      requestRef: this.requestRef,
+      actorRef: this.actorRef.path,
+      offerRef: this.offerRef?.path,
+      requestRef: this.requestRef.path,
```

Comment:
This is part of the challenge using the toObject function. We have multiple opinions in terms of "what is an object".

I'd say right now an "object" is what we get in and out of firestore, which is how it's been implemented everwhere else.

I'd remove the `.path` here.

Reply:
the reason for doing this was because of how the cloud function was sending the data to the frontend, it was being serialized so the data types were getting destroyed.
which is why if you look at the frontend code, appropriate code has been put to reconstruct the data type. but it's not an ideal solution.
this is only for these abstract models and not for the actual model